### PR TITLE
fix: wrong rendering indexes after prepending panels

### LINKED ISSUE #389: Prepending panels with setState can break carousel with renderOnlyVisible option
## Description
Flicking's `getRenderingIndexes` can return the wrong result with non-circular & renderOnlyVisible option enabled.
This issue is reported on the React framework but can occur across all frameworks.

## Steps to check or reproduce
```tsx
<Flicking
  className="flicking flicking0"
  gap={10}
  defaultIndex={1}
  renderOnlyVisible={true}
  onMoveEnd={(e) => {
    if (e.index === 0) {
      this.setState((state) => {
        const start = state.list0[0] || 0;
        return {
          list0: [
            start - 10,
            start - 9,
            start - 8,
            start - 7,
            start - 6,
            start - 5,
            start - 4,
            start - 3,
            start - 2,
            start - 1,
            ...state.list0
          ]
        };
      });
    }
  }}
>
  {this.state.list0.map((num) => {
    return (
      <div
        key={num}
        className={"infinite infinite" + (Math.abs(num) % 5)}
      >
        {num}
      </div>
    );
  })}
</Flicking>
```

### PR DESCRIPTION
**Pull request by WoodNeck**
## Issue
#389 

## Details
This fixes #389

### CODE REVIEW

**Review by mixed — APPROVED** [review]
LGTM

File: src/Flicking.ts [hunk 1]
```diff
@@ -651,7 +652,9 @@ class Flicking extends Component<{
         const cloneIndex = Math.floor(val / prevPanelCount);
```
Comment (daybrush) [inline src/Flicking.ts:652]: `visibleIndexes` is like the latest index. Is it right to use `prevPanelCount`?

If `circular` is used, the `cloneIndex` will be very large.

**Review by daybrush — APPROVED** [review]

File: src/Flicking.ts (hunk 1 above)
Comment (WoodNeck) [reply src/Flicking.ts:652]: `val` is using an absolute index based on previous panel count, see line 645

Comment (daybrush) [reply src/Flicking.ts:652]: In the test, beforeSync was called first, so I thought that the indexes of visiblePanels were all changed.

is correct that cloneIndex and index do not change even after beforeSync?

Comment (WoodNeck) [reply src/Flicking.ts:652]: Well, that should be handled differently. I'll send a PR tomorrow.

**Pull request merged**